#4519: Adding ability to configure custom Authorizer implementation (#4540)
strimzi/strimzi-kafka-operator: documentation/modules/appendix_crds.adoc

|string array
|====

[id='type-KafkaAuthorizationCustom-{context}']
### `KafkaAuthorizationCustom` schema reference

Used in: xref:type-KafkaClusterSpec-{context}[`KafkaClusterSpec`]

xref:type-KafkaAuthorizationCustom-schema-{context}[Full list of `KafkaAuthorizationCustom` schema properties]

include::../api/io.strimzi.api.kafka.model.KafkaAuthorizationCustom.adoc[leveloffset=+1]

[id='type-KafkaAuthorizationCustom-schema-{context}']
==== `KafkaAuthorizationCustom` schema properties

The `type` property is a discriminator that distinguishes use of the `KafkaAuthorizationCustom` type from xref:type-KafkaAuthorizationSimple-{context}[`KafkaAuthorizationSimple`], xref:type-KafkaAuthorizationOpa-{context}[`KafkaAuthorizationOpa`], xref:type-KafkaAuthorizationKeycloak-{context}[`KafkaAuthorizationKeycloak`].
It must have the value `custom` for the type `KafkaAuthorizationCustom`.
[options="header"]
|====
|Property                  |Description
|type               1.2+<.<|Must be `custom`.
|string

> |implementionClass  1.2+<.<|Authorization Implemention class, must be available in classpath.

Reviewer: ```suggestion
|implementionClass  1.2+<.<|Authorization Implementation class, which must be available in classpath.
```

|string

> |superUsers         1.2+<.<|List of super users. Should contain list of user principals which should get unlimited access rights.

Reviewer: ```suggestion
|superUsers         1.2+<.<|List of super users, which are user principals with unlimited access rights.
```
